plugins/jspsych-spr-moving-window.js

```diff
@@ -308,66 +246,47 @@ var sprMovingWindow = (function(jspsych) {
         createCanvas(display_element, trial_pars);
         ctx.font = font;
         let stimulus = trial_pars.stimulus;
-        if (trial_pars.grouping_string) {
-            grouping_re = RegExp(trial_pars.grouping_string, 'ug');
-            groups = createGroups(stimulus, grouping_re);
-            stimulus = stimulus.replace(grouping_re, "");
-        }
-        else {
-            groups = createGroups(stimulus, RE_WHITE_SPACE);
-        }
-        stimulus = stimulus.replace(RegExp("#", 'gu'), "");
-        let lines = stimulus.split(RE_NEWLINE);
-        gatherWordInfo(lines, trial_pars);
+        let parsed_stimulus = parseSpr(stimulus);
+        groups = createGroups(parsed_stimulus);
+        gatherWordInfo(parsed_stimulus, trial_pars);
     }
 
     /**
-     * Create groups of words that are presented together
-     * @param {String} stim the stimulus to be presented
-     * @param {RegExp} split_re
+     * @param {<Array>.<parts.GroupList>} parsed_stim
+     *
+     * @returns {GroupInfo[]}
      */
-    function createGroups(stim, split_re) {
+    function createGroups(parsed_stim) {
```

Comment:
I'm not sure what this function does, as it seems like it's only called once and the return value isn't used

Reply:
It's called in setupVariables() and stores the output in the "global"/plugin scoped private variable groups and this is used in order to draw the groups one by one in the drawStimulus() function.